src/setup_payload/SetupPayload.h

```diff
@@ -270,6 +271,17 @@ class SetupPayload : public PayloadContents
      **/
     static CHIP_ERROR generateRandomSetupPin(uint32_t & setupPINCode);
 
+    /**
+     * @brief Get a list of setup payloads from a string representation.
+     *
+     * @param[in] stringRepresentation The string representing the payloads.
+     *
+     * @return a CHIP_ERROR on failure, a vector of setup payloads on success.  The vector may
+     *         contain only one payload, or may contain multiple payloads if concatenated QR codes
+     *         are used.
+     */
+    static std::variant<CHIP_ERROR, std::vector<SetupPayload>> FromStringRepresentation(std::string stringRepresentation);
```

Comment:
Is this `std::variant<CHIP_ERROR, result type>` a new sanction API pattern? Otherwise we should stick with the existing pattern of returning CHIP_ERROR and having the result as an out parameter.

Reply:
We've been returning variants in some other places....

I _could_ make the vector an outparam, for sure, but it seems like a slightly odd thing for a factory method.

Reply:
It doesn't seem like this kind of API is going to work well with `ReturnErrorOnFailure` and related patterns that are very prevalent in the code base, so to me it feel like we should stick with the current paradigm. I guess the method name should then maybe be something like `populatePayloads`?

Reply:
Will switch to doing a CHIP_ERROR return.